src/test/addrman_tests.cpp

```diff
@@ -62,16 +95,16 @@ BOOST_AUTO_TEST_CASE(addrman_ports)
     // Set addrman addr placement to be deterministic.
     addrman.MakeDeterministic();
 
-    CNetAddr source = CNetAddr("252.2.2.2:8333");
```

Comment:
I did a quick look and didn't find anything other than this:

https://github.com/bitcoin/bitcoin/blob/master/src/netbase.cpp#L622

but it wouldn't surprise me if there was more. I planned on performing a more complete search, especially of the network deserialization, code after I finished these unittests. I'll probably start that tonight.

I'm also considering a mitigation strategy of putting asserts in the constructor since no IP string should contain both "." and ":". There is logic which already does this:

https://github.com/bitcoin/bitcoin/blob/master/src/netbase.cpp#L68

In using asserts I want to be careful not make it worse by allowing crashes. It might be better to have an IP treated as IPv6 than to have a situation where a remote user can crash a bitcoind node. Let me think about it more, but I believe there is a better and less risky strategy than asserts.

Before anyone spends anymore time on this let me write up a test which confirms the behavior and we can continue the discussion on that pull request.